crates/bevy_asset/src/assets.rs

```diff
@@ -270,20 +270,27 @@ pub trait AddAsset {
 }
 
 impl AddAsset for App {
+    /// Add an [`Asset`] to the [`App`].
+    ///
+    /// Adding the same [`Asset`] several time is idempotent.
     fn add_asset<T>(&mut self) -> &mut Self
     where
         T: Asset,
     {
-        let assets = {
-            let asset_server = self.world.get_resource::<AssetServer>().unwrap();
-            asset_server.register_asset_type::<T>()
-        };
+        if !self.world.contains_resource::<Assets<T>>() {
+            let assets = {
+                let asset_server = self.world.get_resource::<AssetServer>().unwrap();
+                asset_server.register_asset_type::<T>()
+            };
 
-        self.insert_resource(assets)
-            .add_system_to_stage(AssetStage::AssetEvents, Assets::<T>::asset_event_system)
-            .add_system_to_stage(AssetStage::LoadAssets, update_asset_storage_system::<T>)
-            .register_type::<Handle<T>>()
-            .add_event::<AssetEvent<T>>()
+            self.insert_resource(assets)
+                .add_system_to_stage(AssetStage::AssetEvents, Assets::<T>::asset_event_system)
+                .add_system_to_stage(AssetStage::LoadAssets, update_asset_storage_system::<T>)
+                .register_type::<Handle<T>>()
+                .add_event::<AssetEvent<T>>()
+        } else {
+            self
```

Comment:
Should we `warn` when an asset is re-added? 

I think the answer is "no" because we are now explicitly allowing this behavior.

Reply:
Yeah, I would prefer not to warn here. It will typically just be noise.

Reply:
I agree, no warn here 🙂